File: src/shmp/visualizer/Turner.java
package shmp.visualizer;

import shmp.simulation.CulturesController;

import java.util.concurrent.atomic.AtomicBoolean;


public class Turner implements Runnable {
    public volatile AtomicBoolean isAskedToStop = new AtomicBoolean(false);
    private int turns;
    private CulturesController controller;
    private int printTurnDelta = 50;

    public Turner(int turns, CulturesController controller) {
        this.turns = turns;
        this.controller = controller;
    }

    @Override
    public void run() {
        for (int i = 0; i < turns; i++) {
            controller.turn();
            if (i % printTurnDelta == 0) {
                CulturesController.visualizer.print();
            }
            if (isAskedToStop.get()) {
                System.out.println("Terminating Turner thread");
                return;
            }
        }
    }
}

File: src/shmp/visualizer/Turner.kt
package shmp.visualizer

import shmp.simulation.Controller
import shmp.simulation.World
import java.util.concurrent.atomic.AtomicBoolean

class Turner(private val turns: Int, private val controller: Controller<out World>) : Runnable {
    @JvmField
    @Volatile
    var isAskedToStop = AtomicBoolean(false)

    private val printTurnDelta = 50

    override fun run() {
        for (i in 0 until turns) {
            controller.turn()
            if (i % printTurnDelta == 0)
                Controller.visualizer.print()

            if (isAskedToStop.get()) {
                println("Terminating Turner thread")
                return
            }
        }
    }
}